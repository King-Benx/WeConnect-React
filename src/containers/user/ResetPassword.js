import React from 'react';
import reset_password from '../../images/reset_password.png';
import DashboardNavigation from '../../navigation/DashboardNavigation';
import CustomFunctions from '../../custom/CustomFunctions';
import { BASE_URL } from '../../custom/constants';
import { Image, Button } from 'react-bootstrap'
import superagent from 'superagent'

class Dashboard extends React.Component{
    constructor(){
        super()
        this.state={
            password:"",
            confirm_password: ""
        }
    }

    handleReset = (event) => {
        this.setState({password:"", confirm_password:""});
    }

    handleChange = (event) => {
        this.setState({[event.target.id]: event.target.value});
    }

    formSubmit = (event) => {
        event.preventDefault();
        if (this.state.password === this.state.confirm_password){
            if (this.state.password.length > 4){
                superagent
                    .post(BASE_URL+'api/v1/auth/reset-password')
                    .send({ new_password:this.state.password })
                    .set({'x-access-token':CustomFunctions.getToken()})
                    .end((err,res)=>{
                        if(err){ CustomFunctions.createNotifications(err.status,err.toString()) }; 
                        CustomFunctions.createNotifications(res.status,res.body.message);
                        this.props.history.replace('/dashboard');
                    });
                }else{
                    CustomFunctions.createNotifications(400,'Password too short!');
                }
            }else{
                CustomFunctions.createNotifications(400,'Passwords are not similar');
        }
    }

    render(){
        return ( 
            <div className="row"> 
                <DashboardNavigation/>
                <div className="col-sm-9 content-wrapper">
                    <div className="row">
                        <div className="col-xs-12 col-sm-4 text-center">
                            <Image src={reset_password} rounded responsive/>
                            <em class="text-info">Secure your information by changing your password regularly</em>
                        </div>
                        <div className="col-xs-12 col-sm-8">
                            <div className="panel panel-default">
                                <div className="panel-body">
                                    <form method="POST" className="form" onSubmit={this.formSubmit}>
                                        <legend className="text text-center">Reset Password</legend>
                                        <div className="form-group">
                                            <input type="password" className="form-control" placeholder="Password" id="password" value={this.state.password}  name="password" onChange={ this.handleChange } aria-describedby="password-addon" required="required"/>
                                        </div>
                                        <div className="form-group">
                                            <input type="password" className="form-control" placeholder="Confirm Password" value={this.state.confirm_password} id="confirm_password"  onChange={ this.handleChange }  name="confirm_password" aria-describedby="confirm_password-addon"
                                                    required="required"/>
                                        </div>
                                        <div className="form-group">
                                            <Button type="reset" onClick={this.handleReset} className="pull-left"><i className="glyphicon glyphicon-refresh"></i> Reset</Button>
                                            <Button type="submit" className="pull-right" bsStyle="warning"><i className="glyphicon glyphicon-wrench"></i> Reset Password</Button>
                                            <div className="clearfix"></div>
                                        </div>
                                    </form>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default Dashboard;