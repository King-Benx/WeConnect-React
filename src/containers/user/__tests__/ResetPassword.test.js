import React from 'react';
import { shallow } from 'enzyme';
import ResetPassword from "../ResetPassword";

describe('These are tests for the login form ', () =>{
    const props = {
        match: {
            params: {

            }
        }
    }
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        }
    }

    it('handles input changed', () => {
        const wrapper = shallow(<ResetPassword />); 
        wrapper.instance().handleChange(event)
    })

    it('handles input reset', () => {
        const wrapper = shallow(<ResetPassword />); 
        wrapper.instance().handleReset(event)
    })

    it('handles submit form', () => {
        const wrapper = shallow(<ResetPassword />); 
        wrapper.instance().formSubmit(event)
    })


    it('renders without crashing', () => {
        shallow(<ResetPassword  />)
    });

    // it('has an image', () => {
    //     const wrapper = shallow(<ResetPassword />)
    //     expect(wrapper.find('Image').length).toBe(1)
    // });

    // it('renders the form', () => {
    //     const wrapper = shallow(<ResetPassword />)
    //     expect(wrapper.find('form').length).toBe(1)
    // });

    // it('renders form input', () => {
    //     const wrapper = shallow(<ResetPassword/>)
    //     expect(wrapper.find('input').length).toBe(2)
    // });

    // it('has divs', () => {
    //     const wrapper = shallow(<ResetPassword />)
    //     expect(wrapper.find('div').length).toBe(6)
    // });

    // it('has buttons', () => {
    //     const wrapper = shallow(<ResetPassword />)
    //     expect(wrapper.find('Link').length).toBe(2)
    // });

    // it('has buttons', () => {
    //     const wrapper = shallow(<LoginPage />)
    //     expect(wrapper.find('Button').length).toBe(2)
    // });
});