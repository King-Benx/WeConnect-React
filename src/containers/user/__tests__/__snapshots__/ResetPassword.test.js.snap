// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the login form  renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9 content-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-4 text-center"
      >
        <img
          className="img-responsive img-rounded"
          src="reset_password.png"
        />
        <em
          class="text-info"
        >
          Secure your information by changing your password regularly
        </em>
      </div>
      <div
        className="col-xs-12 col-sm-8"
      >
        <div
          className="panel panel-default"
        >
          <div
            className="panel-body"
          >
            <form
              className="form"
              method="POST"
              onSubmit={[Function]}
            >
              <legend
                className="text text-center"
              >
                Reset Password
              </legend>
              <div
                className="form-group"
              >
                <input
                  aria-describedby="password-addon"
                  className="form-control"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required="required"
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  aria-describedby="confirm_password-addon"
                  className="form-control"
                  id="confirm_password"
                  name="confirm_password"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required="required"
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <button
                  className="pull-left btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="reset"
                >
                  <i
                    className="glyphicon glyphicon-refresh"
                  />
                   Reset
                </button>
                <button
                  className="pull-right btn btn-warning"
                  disabled={false}
                  type="submit"
                >
                  <i
                    className="glyphicon glyphicon-wrench"
                  />
                   Reset Password
                </button>
                <div
                  className="clearfix"
                />
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
